cmd/kubeadm/app/phases/upgrade/health.go

```diff
@@ -162,8 +186,8 @@ func createJob(client clientset.Interface, cfg *kubeadmapi.ClusterConfiguration)
 	}
 
 	// Wait for the Job to complete
-	err = wait.PollUntilContextTimeout(ctx, time.Second*1, timeout, true, func(ctx context.Context) (bool, error) {
-		job, err := client.BatchV1().Jobs(ns).Get(ctx, jobName, metav1.GetOptions{})
+	err = wait.PollUntilContextTimeout(ctx, time.Second*1, timeout, true, func(_ context.Context) (bool, error) {
+		job, err := client.BatchV1().Jobs(ns).Get(context.Background(), jobName, metav1.GetOptions{})
```

Comment:
why not reuse the ctx parameter?

Reply:
it stomps the original last error with a timeout error. we actually had some more serious problems with this in other areas of kubeadm.

on timeout:

instead of:
`failed to do something: actual API server error`

what happens is:
`failed to do something: timeout after N seconds`